ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/gui/JailPanel.java

```diff
@@ -0,0 +1,72 @@
+package com.drewhannay.chesscrafter.gui;
+
+import com.drewhannay.chesscrafter.dragNdrop.DropManager;
+import com.drewhannay.chesscrafter.dragNdrop.GlassPane;
+import com.drewhannay.chesscrafter.dragNdrop.MotionAdapter;
+import com.drewhannay.chesscrafter.models.*;
+import com.drewhannay.chesscrafter.utility.GuiUtility;
+import com.google.common.collect.Lists;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+public class JailPanel extends JPanel {
+
+    public JailPanel(int totalPieces, int teamIndex) {
+        int jailDimension = (totalPieces / 4) + (totalPieces % 4);
+        setOpaque(false);
+        setLayout(new GridLayout(jailDimension, jailDimension));
+        setPreferredSize(new Dimension(jailDimension * 48, jailDimension * 48));
+        mSquareLabels = new SquareJLabel[jailDimension][jailDimension];
+        mTeamIndex = teamIndex;
+        setBorder(GuiUtility.createBorder(Messages.getString("PlayGamePanel.capturedPieces")));
+        setLayout(new GridLayout(jailDimension, jailDimension));
+        setPreferredSize(new Dimension((jailDimension + 1) * 25, (jailDimension + 1) * 25));
+        createGrid(jailDimension);
+    }
+
+    private void createGrid(int jailDimensions) {
+        for (int x = 1; x <= jailDimensions; x++) {
+            for (int y = 1; y <= jailDimensions; y++) {
+                SquareJLabel square = new SquareJLabel(BoardCoordinate.at(x, y), true, 25);
+                add(square);
+                mSquareLabels[x - 1][y - 1] = square;
+            }
+        }
+    }
+
+    //TODO update the square of the piece in check when Logic can provide it
+    /*public void updateCheckLocation(Board[] boards){
+
+    }*/
+
+    public void updateJailPopulation(Team[] teams) {
```

Comment:
Also, thinking about it more, can we pass in the List of captured pieces here instead of the Team[]?